lms/djangoapps/course_blocks/transformers/hidden_content.py

```diff
@@ -38,10 +40,20 @@ def name(cls):
         return "hidden_content"
 
     @classmethod
+    def _get_merged_self_paced(cls, block_structure, block_key):
```

Comment:
self-paced and course-end are properties on the course - not on each block.

So there's no reason to capture `self_paced` and "end_date" as "merged" fields, which need to be merged throughout the block hierarchy.  

Rather, simply add the following line in the `collect` method and they will be automatically collected for the course block:
```
block_structure.request_xblock_fields(u'self_paced', u'end')
```